react/features/screenshot-capture/ScreenshotCaptureSummary.tsx

```diff
@@ -121,28 +121,34 @@ export default class ScreenshotCaptureSummary {
      * @returns {void}
      */
     stop() {
-        this._streamWorker.postMessage({ id: CLEAR_INTERVAL });
+        this._streamWorker.postMessage({ id: CLEAR_TIMEOUT });
     }
 
     /**
-     * Method that is called as soon as the first frame of the video loads from stream.
-     * The method is used to store the {@code ImageData} object from the first frames
-     * in order to use it for future comparisons based on which we can process only certain
-     * screenshots.
+     * Sends to worker the imageBitmap for the next timeout.
      *
-     * @private
-     * @returns {void}
+     * @returns {Promise<void>}
      */
-    async _initScreenshotCapture() {
-        const imageBitmap = await this._imageCapture.grabFrame();
+    async sendTimeout() {
+        let imageBitmap: ImageBitmap | undefined;
 
-        this._currentCanvasContext?.drawImage(imageBitmap, 0, 0, this._streamWidth, this._streamHeight);
-        const imageData = this._currentCanvasContext?.getImageData(0, 0, this._streamWidth, this._streamHeight);
+        if (!this._imageCapture.track || this._imageCapture.track.readyState !== 'live') {
```

Comment:
@saghul here is the check for the track associated to the image capture so see if it in a state that it cannot grab the frame

Reply:
YEs but why are we sending it anyway?

Reply:
https://w3c.github.io/mediacapture-image/#dom-imagecapture-grabframe

Reply:
Sorry, I feel like I'm not explaining myself well :-) My question is: why would that be relevant for the worker to continue? Why can't it simply react to receiving a message when there is an image and just sit idle in the meantime?

Reply:
the worker serves two roles here, as a timer and as an image checking and blob conversion. When you send an image bitmap to the worker it will process it and then send back a timeout tick with the blob if there is some, when you do not send the bitmap just the timeout it will send back only the timeout tick. If I do not sent it back the timeout back to the worker the will be no time, and maybe I would need to use an external interval

Reply:
Got it!